### PR TITLE
Main branch edge to edge implementation

### PR DESCRIPTION
**Pull request by the-scrambler**
Merge pending codelab update.

### CODE REVIEW

**Comment by android-dev-lxl** [comment]
Please remove this from the gradle file and include as needed. 

       ` freeCompilerArgs += "-opt-in=androidx.compose.material3.ExperimentalMaterial3Api"`

File: app/src/main/java/com/example/dessertrelease/ui/DessertReleaseScreen.kt [hunk 1]
```diff
@@ -108,12 +108,17 @@ private fun DessertReleaseScreen(
 }
```
Comment (android-dev-lxl) [inline app/src/main/java/com/example/dessertrelease/ui/DessertReleaseScreen.kt:108]: You should always set the inner padding to the content padding for lists. Please do the same for Grid layout also.
 ```
            modifier = Modifier
                    .padding(
                        vertical = dimensionResource(R.dimen.padding_medium),
                        horizontal = dimensionResource(R.dimen.padding_medium)
                    )
                    .fillMaxWidth(),
                contentPadding = innerPadding
            )
```

File: app/src/main/java/com/example/dessertrelease/ui/DessertReleaseScreen.kt [hunk 2]
```diff
@@ -108,12 +108,17 @@ private fun DessertReleaseScreen(
...
+) {
     LazyColumn(
         modifier = modifier,
         contentPadding = PaddingValues(
-            vertical = dimensionResource(R.dimen.padding_medium),
-            horizontal = dimensionResource(R.dimen.padding_medium)
+            top = contentPadding.calculateTopPadding() + dimensionResource(R.dimen.padding_medium),
+            bottom = contentPadding.calculateBottomPadding() + dimensionResource(R.dimen.padding_medium),
+            start = dimensionResource(R.dimen.padding_medium),
+            end = dimensionResource(R.dimen.padding_medium)
```
Comment (android-dev-lxl) [inline app/src/main/java/com/example/dessertrelease/ui/DessertReleaseScreen.kt:121]: If you set the required padding to the modifier you can just set scaffold padding to the content padding.
`contentPadding = contentPadding,`

**Review by android-dev-lxl — CHANGES_REQUESTED** [review]
Inner padding is not set correctly. 


Before
image

With Edge to Edge:
image

**Review by android-dev-lxl — APPROVED** [review]

**Review by johnshea — CHANGES_REQUESTED** [review]
Navigation bar on API 33 is not translucent but should be.

Upgrade gradle wrapper to `8.2`
Upgrade Compose compiler to `1.5.1`
Upgrade BOM to `2023.08.00`
Project `build.gradle.kts` - upgrade `plugin` versions to the lastest. (Current latest is `1.8.0` and `1.9.0`)

Screenshot 2023-08-14 at 10 33 39 AM

**Review by johnshea — CHANGES_REQUESTED** [review]
Nav bar is not translucent but it should be.

Screenshot 2023-08-17 at 2 29 47 AM

**Review by johnshea — CHANGES_REQUESTED** [review]
API 33 GridLayout - navigation bar is not translucent but it should be.

Screenshot 2023-08-21 at 11 30 30 AM

**Review by android-dev-lxl — APPROVED** [review]

**Review by johnshea — APPROVED** [review]
lgtm with 2 nits.

**Pull request merged**